torch/csrc/lazy/ts_backend/ts_eager_fallback.cpp

```diff
@@ -149,7 +149,13 @@ void ltc_eager_fallback(
       op, stack, torch::lazy::getBackend()->EagerFallbackDeviceType());
 }
 
-TORCH_LIBRARY_IMPL(_, Lazy, m) {
+void register_ts_ltc_eager_fallback() {
+  // Most backends use TORCH_LIBRARY_* macros which perform their dispatcher
+  // registrations at static library init time, but the lazy Torchscript backend
+  // does not since it is built in the main torch lib but not always used.
+  // In particular, if another external backend wants to register itself to the
+  // same key (Lazy), Torchscript backend must not be initialized.
+  auto m = MAKE_TORCH_LIBRARY_IMPL(_, Lazy);
```

Comment:
ok great, before I just commit this, let me make sure it makes sense to me that we have 2 separate globals declared here. I guess that is equivalent to what the original code did, but I didn't understand that either.

I'm also wondering what the difference between (aten, Lazy) and (_, Lazy) is: I believe I was told the latter is how you set up a 'catchall' which is what we want for our fallback, but its a little funny that 'aten' isn't also 'catchall' since all the ops are presumably in that namespace.

cc @ezyang @bdhirsh

Reply:
Yep, you need both of those blocks. The one in the codegen is to manually register the lazy kernel for every LTC-supported operator. And the one here is to register the Lazy-to-cpu boxed fallback.

The reason for the `_` instead of `aten` is because boxed fallbacks are meant to work "on all operators". So we have a bunch of operators under the `aten` namespace (basically everything in native_functions.yaml), but external projects are allowed to register their own custom operators to the dispatcher, usually under a different namespace.

The lazy-to-cpu fallback should work on all operators, not just the ones in `aten` (this is true for all boxed fallbacks) - which is what the underscore represents.